File: ccode/objShear/src/test/test-lcat.cpp
#include <stdio.h>
#include <stdlib.h>
#include <vector>
#include "../lcat.h"


int main(int argc, char** argv) {

    if (argc != 2 ) {
        printf("usage: test-lcat filename\n");
        exit(45);
    }

    const char* filename = argv[1];

    std::vector<struct lens> lcat;

    read_lens(filename, lcat);

    print_lens_firstlast(lcat);

    add_lens_distances(100.0, 0.3, lcat);

    print_lens_firstlast(lcat);
}

File: fortran/objshear/test/test-lcat.f90
! vim:set ft=fortran:
program main

    use lenslib
    use cosmolib

    character*256 filename
    type(lens), dimension(:), allocatable :: lenses


    if (iargc() /= 1) then
        print *,"Usage: test-lcat filename"
        stop 45
    end if


    call getarg(1,filename)

    call cosmo_init(100.0_8, 0.3_8, 5_8)

    call read_lens_cat(filename, lenses)

    call add_lens_dc(lenses)

    call print_lens_firstlast(lenses)


end program
